R/utils.R

```diff
@@ -155,12 +172,45 @@ invokeRemote = function(map, method, args = list()) {
     )
   )
 
+  sess <- map$session
   if (map$deferUntilFlush) {
-    map$session$onFlushed(function() {
-      map$session$sendCustomMessage("leaflet-calls", msg)
-    }, once = TRUE)
+    if (compareVersion(as.character(packageVersion("shiny")), "0.12.0.9001") < 0) {
+
+      # See comment on sessionFlushQueue.
+
+      if (!exists(sess$token, sessionFlushQueue)) {
+        # If the current session doesn't have an entry in the sessionFlushQueue,
+        # initialize it with a blank list.
+        assign(sess$token, list(), sessionFlushQueue)
```

Comment:
I just want to make sure you're aware that it's much more efficient to do `sessionFlushQueue[[sess$token]] <- list()`.

In general, functions like get/assign/exists are much slower than `[[`. See this thread if you want to know more: 
  http://r.789695.n4.nabble.com/we-need-an-exists-get-hybrid-td4700333.html